pkg/webhookconfig/registration.go

```diff
@@ -621,32 +666,92 @@ func (wrc *Register) checkEndpoint() error {
 	return err
 }
 
-func (wrc *Register) updateResourceValidatingWebhookConfiguration(nsSelector map[string]interface{}) error {
-	validatingCache, _ := wrc.resCache.GetGVRCache(kindValidating)
+func generateWebhookRule(apiGroups, apiVersions, resources, operationTypes []interface{}) map[string]interface{} {
+	return map[string]interface{}{
+		"operations":  operationTypes,
+		"apiVersions": apiVersions,
+		"apiGroups":   apiGroups,
+		"resources":   resources,
+	}
+}
 
-	resourceValidating, err := validatingCache.Lister().Get(wrc.getResourceValidatingWebhookConfigName())
+func (wrc *Register) checkRuleResources(kind, name string) error {
+	_, webhook, err := wrc.getWebhookFromCache(kind, name)
 	if err != nil {
-		return errors.Wrapf(err, "unable to get validatingWebhookConfigurations")
+		return errors.Wrapf(err, "checkRuleResources fail")
+	}
+
+	rules, ok, err := unstructured.NestedSlice(*webhook, "rules")
+	if err != nil || !ok || len(rules) == 0 {
+		return errors.Wrapf(err, "checkRuleResources no rules")
+	}
+	rule, ok := rules[0].(map[string]interface{})
+	if !ok {
+		return errors.Wrapf(err, "checkRuleResources rule type mismatch got %T", rules[0])
+	}
+	resources, ok, err := unstructured.NestedStringSlice(rule, "resources")
+	if !ok || err != nil || len(resources) == 0 {
+		return errors.Wrapf(err, "checkRuleResources resources")
+	}
+
+	kinds := wrc.policyCache.GetKinds()
+	if len(kinds) != len(resources) {
+		return errors.Wrapf(err, "checkRuleResources len diff")
 	}
 
-	webhooksUntyped, _, err := unstructured.NestedSlice(resourceValidating.UnstructuredContent(), "webhooks")
+	for i := 0; i < len(kinds); i++ {
+		if kinds[i] != resources[i] {
+			return errors.Wrapf(err, "checkRuleResources kind diff")
+		}
+	}
```

Comment:
Is the sort in cache.go to make this check match? Guess that's more efficient than doing some search for each, so long as alphabetical order in `resources` is preserved, which does look to be so

Reply:
yes -- this why the sort

Reply:
i not even sure this is actually called in most recent versions of the code / maybe dead code